src/test/scala/vsys/blockchain/state/opcdiffs/IfOpcDiffTest.scala

```diff
@@ -0,0 +1,84 @@
+    1L, 1, tx).right.get
+
+  property("if opcs") {
+    IfOpcDiff.executeOpcBlock(executionContext,
+      DataEntry(Longs.toByteArray(1), DataType.Amount),
+      Seq.empty) should be (Left(ContractDataTypeMismatch))
+    IfOpcDiff.executeOpcBlock(executionContext,
+      DataEntry(Shorts.toByteArray(0), DataType.OpcBlock),
+      Seq.empty) should be (Right((OpcDiff(Map(),Map(),Map(),Map(),Map(),Map(),Map(),Map()),List())))
+    IfOpcDiff.executeOpcBlock(executionContext,
+      DataEntry(Shorts.toByteArray(1), DataType.OpcBlock),
+      Seq.empty) should be (Left(ContractInvalidOPCData))
+
+    IfOpcDiff.runCondition(
+      DataEntry(Array(1.toByte), DataType.Boolean)
+    ) should be (Right(true))
+    IfOpcDiff.runCondition(
+      DataEntry(Array(0.toByte), DataType.Boolean)
+    ) should be (Right(false))
+    IfOpcDiff.runCondition(
+      DataEntry(Longs.toByteArray(1), DataType.Amount)
+    ) should be (Left(ContractDataTypeMismatch))
+
+    IfOpcDiff.parseBytes(executionContext)(
+      Array[Byte](1.toByte, 0.toByte, 1.toByte),
+      Seq[DataEntry](DataEntry(Array(1.toByte), DataType.Boolean),
+        DataEntry(Shorts.toByteArray(0), DataType.OpcBlock))) should be (
+      Right((OpcDiff(Map(),Map(),Map(),Map(),Map(),Map(),Map(),Map()),
+        List(DataEntry(Array(1.toByte), DataType.Boolean),
+        DataEntry(Shorts.toByteArray(0), DataType.OpcBlock)))))
+    IfOpcDiff.parseBytes(executionContext)(
+      Array[Byte](1.toByte, 0.toByte, 1.toByte),
+      Seq[DataEntry](DataEntry(Array(0.toByte), DataType.Boolean),
+        DataEntry(Shorts.toByteArray(0), DataType.OpcBlock))) should be (
+      Right((OpcDiff(Map(),Map(),Map(),Map(),Map(),Map(),Map(),Map()),
+        List(DataEntry(Array(0.toByte), DataType.Boolean),
+          DataEntry(Shorts.toByteArray(0), DataType.OpcBlock)))))
+    IfOpcDiff.parseBytes(executionContext)(
+      Array[Byte](2.toByte, 0.toByte, 1.toByte, 2.toByte),
+      Seq[DataEntry](DataEntry(Array(1.toByte), DataType.Boolean),
+        DataEntry(Shorts.toByteArray(0), DataType.OpcBlock),
+        DataEntry(Shorts.toByteArray(0), DataType.OpcBlock))) should be (
+      Right((OpcDiff(Map(),Map(),Map(),Map(),Map(),Map(),Map(),Map()),
+        List(DataEntry(Array(1.toByte), DataType.Boolean),
+          DataEntry(Shorts.toByteArray(0), DataType.OpcBlock),
+          DataEntry(Shorts.toByteArray(0), DataType.OpcBlock)))))
+    IfOpcDiff.parseBytes(executionContext)(
+      Array[Byte](2.toByte, 0.toByte, 1.toByte, 2.toByte),
+      Seq[DataEntry](DataEntry(Array(0.toByte), DataType.Boolean),
+        DataEntry(Shorts.toByteArray(0), DataType.OpcBlock),
+        DataEntry(Shorts.toByteArray(0), DataType.OpcBlock))) should be (
+      Right((OpcDiff(Map(),Map(),Map(),Map(),Map(),Map(),Map(),Map()),
+        List(DataEntry(Array(0.toByte), DataType.Boolean),
+          DataEntry(Shorts.toByteArray(0), DataType.OpcBlock),
+          DataEntry(Shorts.toByteArray(0), DataType.OpcBlock)))))
```

Comment:
What's the difference here between if and else part?

Reply:
no difference in `OpcBlock`

Reply:
you can design and let else part return a validation error~ = =\\

Reply:
ok